src/app/ClusterInfo.h

```diff
@@ -44,7 +44,14 @@ struct ClusterInfo
             return true;
         }
 
-        if (mFlags.Has(Flags::kFieldIdValid) && other.mFlags.Has(Flags::kFieldIdValid) && (mFieldId == other.mFieldId))
+        if (mFlags.Has(Flags::kFieldIdValid) && !mFlags.Has(Flags::kListIndexValid) && other.mFlags.Has(Flags::kFieldIdValid) &&
+            (mFieldId == other.mFieldId))
+        {
+            return true;
+        }
+
+        if (mFlags.Has(Flags::kFieldIdValid) && mFlags.Has(Flags::kListIndexValid) && other.mFlags.Has(Flags::kListIndexValid) &&
```

Comment:
These conditions are very hard to follow and generally seem to hinge in mFlags optiosn that are repeated. Could you find a way to make them clearer? Even a comment top of the method explaining what fields exist (endpoint, cluster, field, index) and how they relate to each other may help. 

Maybe locally you could use std::optional to combine mFlags and a value (it has  equality operators and an 'ValueOr' functionality if missing is supposed to be a glob value).

Can a ClusterInfo contain no fieldId but a listid? Why do we check of other.ListValid only but for mFlags we check for both fieldid and listid valid?

Given the complexity of the conditionals here, I have a hard time ensuring that the logic is sane. I would rather expect individual checks of 'fieldid match' then 'listindex match' without combining the two (unless combinining is required).

Reply:
Thinking of logic like:

```
Optional<AttributeId> myFieldId = 
    mFlags.has(Flags::kFieldIdValid) 
        ? Optional<AttributeId>::Value(mFieldId);
        : Optional<AttributeId>::Missing();
       
if (myFieldId == Optional<AttributeId>::Value(0xFFFFFFFF)) {
   return true;  /// TODO: above constant should be a constexpr not hardcoded
}

Optional<AttributeId> otherFieldId = ....;

if (myFieldId != otherFieldId) {
  return false;
}

//// setup myListindex and otherListIndex as Optional<ListIndex>

// either both missing or both exactly the same
return (myListIndex == otherListIndex);

```

However I do not know how to solve a potential no field id and valid index... can indexes exist without a field ID or can  indexes exist with a wildcard field id?

I expect this info in a comment explaining the structure and combination possibilities.

Reply:
Done, thanks